apps/console/src/features/connections/api/authenticators.ts

```diff
@@ -0,0 +1,385 @@
+                throw new IdentityAppsApiException(
+                    ConnectionManagementConstants.MULTI_FACTOR_AUTHENTICATOR_UPDATE_INVALID_STATUS_CODE_ERROR,
+                    null,
+                    response.status,
+                    response.request,
+                    response,
+                    response.config);
+            }
+
+            return Promise.resolve(response.data);
+        }).catch((error: AxiosError) => {
+            throw new IdentityAppsApiException(
+                ConnectionManagementConstants.MULTI_FACTOR_AUTHENTICATOR_UPDATE_ERROR,
+                error.stack,
+                error.code,
+                error.request,
+                error.response,
+                error.config);
+        });
+};
+
+/**
+ * Get Local Authenticator details from `t/<TENANT>>/api/server/v1/configs/authenticators/<AUTHENTICATOR_ID>`
+ *
+ * @param id - Authenticator ID.
+ * @returns Response as a promise.
+ * @throws IdentityAppsApiException
+ */
+export const getLocalAuthenticator = (id: string): Promise<AuthenticatorInterface> => {
+
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: `${ store.getState().config.endpoints.localAuthenticators }/${ id }`
+    };
+
+    return httpClient(requestConfig)
+        .then((response: AxiosResponse<AuthenticatorInterface>) => {
+            if (response.status !== 200) {
+                throw new IdentityAppsApiException(
+                    ConnectionManagementConstants.LOCAL_AUTHENTICATOR_FETCH_INVALID_STATUS_CODE_ERROR,
+                    null,
+                    response.status,
+                    response.request,
+                    response,
+                    response.config);
+            }
+
+            return Promise.resolve(response.data);
+        }).catch((error: AxiosError) => {
+            throw new IdentityAppsApiException(
+                ConnectionManagementConstants.LOCAL_AUTHENTICATOR_FETCH_ERROR,
+                error.stack,
+                error.code,
+                error.request,
+                error.response,
+                error.config);
+        });
+};
+
+/**
+ * Get the details of a Multi-factor authenticator using the MFA connectors in Governance APIs.
+ *
+ * @param id - Authenticator ID.
+ * @returns Response as a promise.
+ * @throws IdentityAppsApiException
+ */
+export const getMultiFactorAuthenticatorDetails = (id: string): Promise<MultiFactorAuthenticatorInterface> => {
+
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: `${ store.getState().config.endpoints.multiFactorAuthenticators }/connectors/${ id }`
+    };
+
+    return httpClient(requestConfig)
```

Comment:
any reason not to use SWR for this new methods?

Reply:
As these changes are directly brought from previous PRs, will continue as it is.